components/Search.tsx

```diff
@@ -33,17 +33,32 @@ const StyledInputBase = styled((props: InputBaseProps) => <OutlinedInput {...pro
 }))
 
 const Search = () => {
-  const { push, asPath } = useRouter()
+  const { push, asPath, query } = useRouter()
   const searchRef = useRef<HTMLInputElement | null>(null)
   const [showClearBtn, setShowClearBtn] = useState(false)
   const isHome = asPath === '/' || asPath === '/zh-CN'
   const theme = useTheme()
 
+  useEffect(() => {
+    if (!searchRef.current) return
+    if (query.search) {
+      searchRef.current.value = (query.search as string) || ''
+    } else {
+      const queryKey = 'search'
+      const queryValue = query[queryKey] || asPath.match(new RegExp(`[&?]${queryKey}=(.*)(&|$)`))
```

Comment:
queries could be decoded with [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)